### PR TITLE
zstbi: Added HDR support to Image.loadFromMemory()

### PR DESCRIPTION
**Pull request by soenkekrusche**
I am fairly new to zig, but used stbi in the past. I am not very deep into the internas of it.
I tried to puzzle together what has to be done with existing code and the stbi.h

I am not sure if I built zstbi successfully as it is a library. I would appreciate if someone could tell me a way to test if my changes do remotely what I think :)

### CODE REVIEW

File: libs/zstbi/src/zstbi.zig [hunk 1]
```diff
@@ -146,14 +146,43 @@ pub const Image = struct {
...
+                &y,
+                &ch,
+                @intCast(c_int, forced_num_components),
+            );
+            if (ptr == null) return error.ImageInitFailed;
+
+            num_components = if (forced_num_components == 0) @intCast(u32, ch) else forced_num_components;
+            width = @intCast(u32, x);
+            height = @intCast(u32, y);
+            bytes_per_component = 1;
```
Comment (michal-z) [inline libs/zstbi/src/zstbi.zig:173]: We use `f16` for each component so `bytes_per_component` should be 2.

**Review by michal-z — CHANGES_REQUESTED** [review]

**Comment by michal-z** [comment]
Thanks for the PR! It looks good. I found one issue/typo.

**Comment by michal-z** [comment]
To test it we would have to create a small test app (or modify existing sample app). This is not strictly necessary as part of this PR.

**Comment by soenkekrusche** [comment]
I will resolve the issue after work and try to write a test or simple test app, no promises for that tho

**Pull request merged**